tests/trainer/loops/test_training_loop.py

```diff
@@ -190,3 +190,14 @@ def training_epoch_end(self, outputs) -> None:
     trainer = Trainer(default_root_dir=tmpdir, fast_dev_run=2)
     trainer.fit(model)
     assert model.on_train_batch_end_called == 2
+
+
+def test_batch_loop_releases_loss(tmpdir):
+    class TestModel(BoringModel):
```

Comment:
```suggestion
def test_batch_loop_releases_loss(tmpdir):
    """Test that loss/graph is released and that it can be garbage collected before the next training step"""
    class TestModel(BoringModel):
```